### PR TITLE
Add Type Specific Empire Properties

### PR DESCRIPTION
**Pull request by jshep23**
Resolves #18 

Added the following type-specific Empire properties:
- EmpireBoolProperty
- EmpireNullableBoolProperty
- EmpireStringProperty
- EmpireNullableStringProperty
- EmpireIntProperty
- EmpireNullableIntProperty
- EmpireDoubleProperty
- EmpireNullableDoubleProperty
- EmpireListProperty
- EmpireMapProperty

### CODE REVIEW

**Review by mdryden — COMMENTED** [review]
Missing docstrings for the typed empire properties

File: lib/empire_exceptions.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+abstract class EmpireException implements Exception {
```
Comment (mdryden) [inline lib/empire_exceptions.dart:1]: Missing docstring

File: lib/empire_exceptions.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  final StackTrace? stack;
+  final String? propertyName;
+  final Type type;
+
+  EmpireException(this.stack, this.propertyName, this.type);
+}
+
+class EmpireNullValueException extends EmpireException {
```
Comment (mdryden) [inline lib/empire_exceptions.dart:9]: Missing docstring

File: examples/counter/lib/counter_view_model.dart [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import 'package:empire/empire_property.dart';
 import 'package:empire/empire_view_model.dart';
 
 class CounterViewModel extends EmpireViewModel {
```
Comment (mdryden) [inline examples/counter/lib/counter_view_model.dart:4]: Maybe the example should be updated to use an EmpireIntProperty?

File: lib/empire_view_model.dart [hunk 4]
```diff
@@ -344,8 +403,57 @@ class EmpireStateChanged<T> {
   final T? previousValue;
   final T? nextValue;
   final String? propertyName;
+  final String? description;
+
+  EmpireStateChanged(this.nextValue, this.previousValue, {this.propertyName, this.description});
+
+  static EmpireStateChanged addedToList<V>(V newValue, {String? propertyName}) =>
```
Comment (mdryden) [inline lib/empire_view_model.dart:410]: Should these have docstrings too?

**Review by mdryden — CHANGES_REQUESTED** [review]

File: examples/counter/lib/counter_view_model.dart (hunk 3 above)
Comment (jshep23) [reply examples/counter/lib/counter_view_model.dart:4]: As you wish my lord

**Review by mdryden — APPROVED** [review]

**Review by mdryden — APPROVED** [review]

**Pull request merged**